docs: Document what MFEs are on by default

feanil opened the pull request:
Document which MFEs are currently on by default in the edx-platform code
and where the code expects those MFEs to be running.

Note: Many others are running by default in Tutor but these are the ones that are on-by default in the edx-platform code as far as I can tell.

feanil committed:
docs: Document what MFEs are on by default

Document which MFEs are currently on by default in the edx-platform code
and where the code expects those MFEs to be running including the
settings that can be overridden to change those default locations.

robrap commented on README.rst:
```diff
@@ -168,9 +168,33 @@ Set up CMS SSO (for Production):
+Expected MFEs and Default Ports
+===============================
+Currently the following MFEs are enabled by default, and are expected to be
+running at the given ports.
```
I'm not certain exactly what this sentence means in edx-platform docs. If I were a newcomer and didn't know what an MFE was, I would probably be even more confused.

bradenmacdonald replied:
Yeah, can you clarify what is meant by "expected"? And if that will change in the future? I would expect for example that this repo eventually becomes headless and doesn't make any expectations at all about what frontends are available at what URLs. With one possible exception being email template defaults ? (Maybe Django admin helpers too ?)

For example, when I look at `COURSE_AUTHORING_MICROFRONTEND_URL` in the codebase, every single occurrence is more or less legacy functionality / something that I think should be removed or refactored.

And I think it would be better to reference the setting names like `COURSE_AUTHORING_MICROFRONTEND_URL` instead of `localhost:2001` as the latter doesn't apply in production, nor tell operators how to configure it if they need a non-default value.

feanil replied:
That's a good call, calling out the settings is useful and I'll add that.  To explain my motivation here a bit, I've been running locally without tutor for a lot of my testing and dev because I find it to be less error prone and one of the issues is knowing which MFEs I need to enable because the default edx-platform code expects them to be on now.

I'll add some more context around that, but my long term goal here is to make it clear what is providing UI and how to get that running on top of the eventually headless edx-platform.  But the more I setup, the more I'm noticing code we haven't cleaned up or defaults that are not correct in platform and are just overridden in tutor.

I want the platform to startup more easily in the default configuration with minimal changes needed by developers and operators.

I'll make some updates based on the feedback here.  Thanks!

bradenmacdonald replied:
>  I've been running locally without tutor for a lot of my testing and dev

Interesting. I'd be curious to hear more about how this works for you. Is it still containerized? And I'm just thinking, with the new frontend-base, if there's just a single repo to run the frontend, then it could become pretty easy to run edx-platform and the new frontend both directly on your system, without tutor or even docker (?).

feanil replied:
Yea, I'm really excited about frontend-base for this reason among many others.

I currently run all the data services I need (sql, redis, mongo) in containers but then just run edx-platform on my host machine.  With the [minimal config](https://github.com/openedx/edx-platform/blob/master/lms/envs/minimal.yml), I can generally run, tests, `runserver` and mfes locally as I need them for whatever I'm working on.

One of the big issues is that tutor does a lot of the "defaulting" internally instead of updating the base config which I'm trying to slowly fix and document the base assumptions here so that they're explicit.

feanil commented:
Hey y'all I've updated the docs, I moved them down to  a section that already provides some relevant context and added the names of the relevant settings to override the defaults.

bradenmacdonald commented on README.rst:
```diff
@@ -185,6 +189,7 @@ This will give you a mostly-headless Open edX platform. Most frontends have
 been migrated to "Micro-Frontends (MFEs)" which need to be installed and run
 separately. At a bare minimum, you will need to run the `Authentication MFE`_,
```
This sentence says you need the Auth MFE, so should it not be listed in the table below? Same question about Learner Dashboard which is also mentioned here.

feanil replied:
Dashboard is needed but it turns out that AuthN is not on by default yet. I've updated the quick paragraph to be more correct and added the learner dashboard to the table below.
